sound/soc/sof/sof-client.h

```diff
@@ -0,0 +1,78 @@
+/* SPDX-License-Identifier: (GPL-2.0 OR BSD-3-Clause) */
+/*
+ * This file is provided under a dual BSD/GPLv2 license.  When using or
+ * redistributing this file, you may do so under either license.
+ *
+ * Copyright(c) 2020 Intel Corporation. All rights reserved.
+ *
+ * Author: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>
+ */
+
+#ifndef __SOUND_SOC_SOF_CLIENT_H
+#define __SOUND_SOC_SOF_CLIENT_H
+
+#include <linux/device/driver.h>
+#include <linux/kernel.h>
+#include <linux/list.h>
+#include <linux/virtual_bus.h>
+#include "sof-priv.h"
+
+enum sof_client_type {
+	SOF_CLIENT_AUDIO,
+	SOF_CLIENT_IPC,
+};
+
+/* SOF client device */
+struct sof_client_dev {
+	struct virtbus_device vdev;
+	void *data;
+	struct list_head list;	/* item in SOF core client drv list */
```

Comment:
would it be useful to have a probe_completion and probe_complete boolean here, set when the client driver completes its probe? It was one of the misses with the platform devices, we could never tell if a machine driver failed to probe whereas with a wait_for_completion it's quite easy.

see what we did for SoundWire,

Reply:
good idea. I'll add that